src/browser/session.ts

```diff
@@ -127,6 +121,41 @@ class Session extends EventEmitter {
             type: 'idle-state-changed'
         });
     }
+
+    /**
+     * Send out an event that lets subscribers know that the session state
+     * of the machine has been changed
+     */
+    private fireSessionEvent(reason: string): void {
+        this.emit('session-changed', {
+            reason,
+            topic: 'system',
+            type: 'session-changed'
+        });
+    }
+
+    /**
+     * deal with the lock event
```

Comment:
Nit: this comment is no longer entirely accurate since this function now handles both lock and unlock events